app/routes/login.tsx

```diff
@@ -24,107 +23,72 @@ export const links: LinksFunction = () => {
         <h1>Login</h1>
-        <Form method="post">
-          <fieldset>
-            <legend className="sr-only">Login or Register?</legend>
-            <label>
-              <input
-                type="radio"
-                name="loginType"
-                value="login"
-                defaultChecked={
-                  !actionData?.fields?.loginType ||
-                  actionData?.fields?.loginType === "login"
-                }
-              />{" "}
-              Login
-            </label>
-            <label>
-              <input
-                type="radio"
-                name="loginType"
-                value="register"
-                defaultChecked={actionData?.fields?.loginType === "register"}
-              />{" "}
-              Register
-            </label>
-          </fieldset>
-          <div>
-            <label htmlFor="username-input">Username</label>
-            <input
-              type="text"
-              id="username-input"
-              name="username"
-              defaultValue={actionData?.fields?.username}
-              aria-invalid={fieldHasErrors(actionData, "username")}
-              aria-errormessage={
-                fieldHasErrors(actionData, "username")
-                  ? "username-error"
-                  : undefined
-              }
-            />
-            {fieldHasErrors(actionData, "username") ? (
-              <p
-                className="form-validation-error"
-                role="alert"
-                id="username-error"
-              >
-                {fieldFirstMessage(actionData, "username")}
-              </p>
-            ) : null}
-          </div>
-          <div>
-            <label htmlFor="password-input">Password</label>
-            <input
-              id="password-input"
-              name="password"
-              defaultValue={actionData?.fields?.password}
-              type="password"
-              aria-invalid={fieldHasErrors(actionData, "password")}
-              aria-errormessage={
-                fieldHasErrors(actionData, "password")
-                  ? "password-error"
-                  : undefined
-              }
-            />
-            {fieldHasErrors(actionData, "password") ? (
-              <p
-                className="form-validation-error"
-                role="alert"
-                id="password-error"
-              >
-                {fieldFirstMessage(actionData, "password")}
-              </p>
-            ) : null}
-          </div>
-          <div id="form-error-message">
-            {actionData?.errors.length ? (
-              <p className="form-validation-error" role="alert">
-                {actionData.errors.map(({ message }) => message).join(", ")}
-              </p>
-            ) : null}
-          </div>
-          <button type="submit" className="button">
-            Submit
-          </button>
+        <Form schema={registrationSchema} hiddenFields={["redirectTo"]}>
```

Comment:
You still need to render the `redirectTo` field inside the Form's children. Whenever you pass children to Form, it doesn't render any field automatically for you.

You also need to pass the value to it. When you do it, I think it's more elegant to no use the `hiddenFields` prop and do `<Field name="redirectTo" value={yourValue} hidden />`

Reply:
Oh, great! That's exactly my intuition - having to place the Field component - but I must have forgotten last night